services/file_service.py

```
@@ -283,3 +283,61 @@ def cleanup_old_uploads(days: int = 7) -> int:
     return removed
+
+
+def read_file_content(file_path: str) -> Optional[str]:
+    """
+    Reads the content of a file, attempting to detect encoding if necessary.
+    Handles various path edge-cases and raises FileReadError for issues.
+    """
+    normalized_path = Path(file_path).resolve()
+
+    if not normalized_path.exists():
+        raise FileReadError(f"File not found: {file_path}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)